### PR TITLE
Paul's branch

### PR DESCRIPTION
**Pull request by paul-crowdbotics**
This contains all of the latest changes.

### CODE REVIEW

File: cookiecutter.json [hunk 1]
```diff
@@ -1,18 +1,23 @@
 {
-    "project_slug": "",
+    "project_slug": "sample",
```
Comment (charathram) [inline cookiecutter.json:2]: Shouldn't this be the `cookiecutter.project_slug` variable?

File: cookiecutter.json [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
     "snake_case_slug": "{{ cookiecutter.project_slug|lower|replace(' ', '_')|replace('-', '_') }}",
     "dash_case_slug": "{{ cookiecutter.snake_case_slug|replace('_', '-') }}",
     "github_url": "https://github.com/crowdbotics-users/{{ cookiecutter.dash_case_slug }}",
-    "custom_domain": "",
-    "has_devtools": "y",
+    "custom_domain": "crowdbotics",
+    "custom_domain_extension": "com",
+    "custom_project_package": "{{ cookiecutter.project_slug|lower|replace(' ', '')|replace('-', '') }}",
 	"has_h2_database": "y",
-	"has_jpa": "y",
```
Comment (charathram) [inline cookiecutter.json:10]: Why are we removing `has_jpa`?

Comment (paul-crowdbotics) [reply cookiecutter.json:10]: As part of the requirements:

Default Spring components (cannot be changed): 
Spring Web starter
**Spring Data JPA starter**
Spring Security

Comment (charathram) [reply cookiecutter.json:10]: Oh. Got it. However, I think Mike (from USAF) said he wanted this to be an option. Let me read his email once again.

File: cookiecutter.json (hunk 1 above)
Comment (paul-crowdbotics) [reply cookiecutter.json:2]: Are you asking about the initialization with the literal "sample"?

Comment (charathram) [reply cookiecutter.json:2]: Yes. I am. I'm not certain that should be hardcoded. @driverdan might have an opinion.

Comment (paul-crowdbotics) [reply cookiecutter.json:2]: I did that for testing purposes knowing that it should be overridden by the web app.

File: cookiecutter.json (hunk 2 above)
Comment (charathram) [reply cookiecutter.json:10]: Nah. I was wrong. Let's make this included by default.

File: cookiecutter.json (hunk 1 above)
Comment (driverdan) [reply cookiecutter.json:2]: In this case "sample" would be used as a default value if `project_slug` isn't passed in at runtime.

File: cookiecutter.json [hunk 3]
```diff
@@ -1,18 +1,23 @@
...
 	"has_mysql_database": "n",
 	"has_postgresql_database": "n",
 	"has_redis_database": "n",
+	"has_spring_boot_devtools": "y",
+	"has_spring_boot_ldap": "n",
+	"has_spring_boot_rest": "n",
+	"has_spring_boot_session": "n",
+	"entity_id_type": "String",
 	"java_version": "1.8",
 	"spring_boot_version": "2.1.9.RELEASE"
```
Comment (driverdan) [inline cookiecutter.json]: Looks like some of this indentation doesn't match. Maybe accidentally tabs instead of spaces?

**Review by driverdan — CHANGES_REQUESTED** [review]

File: cookiecutter.json (hunk 1 above)
Comment (charathram) [reply cookiecutter.json:2]: I think that's fine, right?

Comment (crowdbotics-apps) [reply cookiecutter.json]: If that's what we want as a default it's fine. It should always be overridden when created from the dashboard.

Comment (driverdan) [reply cookiecutter.json:2]: If that's what we want as a default it's fine. It should always be overridden when created from the dashboard.

File: cookiecutter.json (hunk 3 above)
Comment (paul-crowdbotics) [reply cookiecutter.json]: The use of spaces instead of tabs is required?

**Pull request closed without merging**